### PR TITLE
[widget audit] toga.DateInput, toga.TimeInput

### LINKED ISSUE #1962: `avc: denied { ioctl }` messages in Android testbed
The Android testbed produces a message like this approximately once per test:

`W/Thread-2: type=1400 audit(0.0:767): avc: denied { ioctl } for path="pipe:[10138300]" dev="pipefs" ino=10138300 ioctlcmd=0x5413 scontext=u:r:untrusted_app:s0:c162,c256,c512,c768 tcontext=u:r:untrusted_app:s0:c162,c256,c512,c768 tclass=fifo_file permissive=0 app=org.beeware.toga.testbed`

This is an SELinux message, probably caused by the pytest-asyncio plugin, which I think uses a pipe as a means of thread synchronization. 

It doesn't appear to be causing any problems, it just makes the log harder to read. But I don't want to suppress all "avc: denied" messages, because they often contain useful information about why the app has been blocked from doing something.

### PR DESCRIPTION
**Pull request by freakboy3742**
Audit of the DateInput and TimeInput widgets.

Renames the widgets from Date/TimePicker to Date/TimeInput for consistency. The widget was previously undocumented, and is only currently implemented for Android and Windows, so the impact of this change should be minimal. A compatibility shim for the old name has been included.

Fixes #1962.

## Audit checklist
* [x]  Core tests ported to pytest
* [x]  Core tests at 100% coverage
* [x]  Core docstrings complete, and in Sphinx format
* [x]  Documentation complete and consistently formatted
* [x]  cocoa backend at 100% coverage (not implemented)
* [x]  winforms backend at 100% coverage
* [x]  gtk backend at 100% coverage (not implemented)
* [x]  iOS backend at 100% coverage (not implemented)
* [x]  android backend at 100% coverage
* [x]  Widget support table updated
* [x]  Widget Issue backlog reviewed

### CODE REVIEW

**Comment by mhsmith** [comment]
> Renames the widgets from Date/TimePicker to Date/TimeInput for consistency. The widget was previously undocumented, and is only currently implemented for Android and Windows, so the impact of this change should be minimal. A compatibility shim for the old name has been included.

For consistency with NumberInput, I've also renamed `min_date` and `max_date` to `min_value` and `max_value`, and similarly with the TimeInput. At some point we might want to do the same thing with the Slider, which is an outlier because it uses a combined `range` rather than separate min and max.

**Comment by freakboy3742** [comment]
> For consistency with NumberInput, I've also renamed `min_date` and `max_date` to `min_value` and `max_value`, and similarly with the TimeInput. 

Good catch - 100% agreed.

> At some point we might want to do the same thing with the Slider, which is an outlier because it uses a combined `range` rather than separate min and max.

Also agreed. 

It occurs to me that we should start logging specific tickets for the resolveable issues that we're finding as a result of this audit (e.g., styling of Selection on GTK, ActivityIndicator not implemented on Informs, etc). Prior to the audit, there wasn't a whole lot of value to having tickets for all the micro issues and missing platform features; now that we have coverage testing, we have a specific list of things that are missing, the platforms where they're missing, and a known contract that they should be adhering to. This sort of API inconsistency fits that pattern.

**Comment by mhsmith** [comment]
@freakboy3742: The Android testbed is hanging intermittently, so this isn't quite ready to merge, but it would be useful for you to review it now.

File: winforms/src/toga_winforms/widgets/dateinput.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import datetime
+
+from travertino.size import at_least
+
+from toga_winforms.libs import WinDateTime, WinForms
+
+from .base import Widget
+
+NO_MIN = WinForms.DateTimePicker.MinDateTime  # Year 1753
+NO_MAX = WinForms.DateTimePicker.MaxDateTime  # Year 9998
```
Comment (freakboy3742) [inline winforms/src/toga_winforms/widgets/dateinput.py:10]: Worth documenting these limits in the user-facing notes.

**Review by freakboy3742 — COMMENTED** [review]
Periodic freezing notwithstanding, this looks pretty good; I've flagged a couple of things we should probably raise in the user-facing notes for the widgets, plus one possible approach for a "soft" min/max on Android TimeInput.

File: android/src/toga_android/widgets/dateinput.py [hunk 2]
```diff
@@ -7,78 +7,67 @@
 )
 from .internal.pickers import PickerBase
 
+NO_MIN = date(1799, 1, 1)
+NO_MAX = date(9999, 1, 1)
```
[the remark this answers is not in the document]
Comment (mhsmith) [reply android/src/toga_android/widgets/dateinput.py]: As discussed, I've changed the implementation to allow years between 1800 and 8999 on all platforms, and updated the documentation to match.

File: winforms/src/toga_winforms/widgets/dateinput.py (hunk 1 above)
Comment (mhsmith) [reply winforms/src/toga_winforms/widgets/dateinput.py]: See above.

**Comment by mhsmith** [comment]
I've changed the min and max properties as follows:

* They always return an actual date or time rather than None.

* Consider changing the min/max from 10/20 to 30/40. Previously, to avoid a ValueError, you would need to update the max before the min. But when going from 30/40 to 10/20, you would have to set the min before the max. This would make it inconvenient to set arbitrary limits. So instead of throwing a ValueError, the min can now clip the max and vice versa.

**Review by mhsmith — APPROVED** [review]
Ready to merge.

**Review by freakboy3742 — COMMENTED** [review]
👍

**Pull request merged**